datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -765,35 +909,38 @@ mod tests {
             list,
             &true,
             vec![Some(false), Some(true), None],
-            col_a.clone()
+            col_a.clone(),
+            &schema
         );
 
         // expression: "a in (0, 1, NULL)"
         let list = vec![
             lit(ScalarValue::Int64(Some(0))),
             lit(ScalarValue::Int64(Some(1))),
-            lit(ScalarValue::Utf8(None)),
+            lit(ScalarValue::Null),
```

Comment:
We can't get the coerced data type between INT64 and UTF8.
I think the type of NULL is  `Null`

Reply:
From this test case, we can know that we should not treat Null as UTF8 data type